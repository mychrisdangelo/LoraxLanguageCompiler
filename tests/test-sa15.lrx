/*
 * Author: Chris D'Angelo
 * Testing valid semantic analysis
 */

int main()
{
	int n;
	tree <int> t(2);
	tree <char> t2(4);
	tree <int> t3(n);
	
	null == null;
	t == null;
	t + t == t3;
	t != t2;

	return 0;
}