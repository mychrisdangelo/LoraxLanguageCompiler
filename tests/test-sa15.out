Warning: comparison of tree <int>(2) with type tree <char>(4)Passed Semantic Analysis.
